### PR TITLE
Cart context

### PR DESCRIPTION
**Pull request by spraveenofficial**
Added Cart Context.

Initial State managing by Existing LocalStorage cart key.

Added Reducer for Managing Cart.

### CODE REVIEW

**Review by AmishMishra11 — COMMENTED** [review]
- clean code
- good reducer implementation

File: frontend/src/Constants/cart.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const ADD_TO_CART = "ADD_TO_CART";
```
Comment (AmishMishra11) [inline frontend/src/Constants/cart.js:1]: Can you tell me why did you do it this way instead of using them directly in the reducer functiuon?

**Review by AmishMishra11 — COMMENTED** [review]
- clean code
- good reducer implementation

File: frontend/src/Constants/cart.js (hunk 1 above)
Comment (spraveenofficial) [reply frontend/src/Constants/cart.js:1]: It allows you to easily find all usages of that constant across the project. It also prevents you from introducing silly bugs caused by typos -- in which case, you will get a ReferenceError immediately.

File: frontend/src/Contexts/cart-context.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const [state, dispatch] = useReducer(cartReducer, {
+    cart: localStorage.getItem("cart")
+      ? JSON.parse(localStorage.getItem("cart"))
+      : [],
+    wishList: localStorage.getItem("wishlist")
+      ? JSON.parse(localStorage.getItem("wishlist"))
+      : [],
+    totalItems: localStorage.getItem("cart")
+      ? JSON.parse(localStorage.getItem("cart")).reduce(
+          (a, b) => Number(a) + Number(b.quantity),
```
Comment (ArjunGTX) [inline frontend/src/Contexts/cart-context.js]: These variables could have been named better

**Review by mnath30 — APPROVED** [review]
Code looks clean
Liked the part where you have defined the constants separately

File: frontend/src/Hooks/carousel.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    items: name,
+  };
+};
+
+// Hook for setting aka Next & Previous for Home Corousel
+const setCorousel = () => {
+  let gap = 16;
+  const nextFunction = () => {
+    const carousel = document.querySelector(".carousel-main"),
+      content = document.querySelector(".carousel-content"),
```
Comment (ArjunGTX) [inline frontend/src/Hooks/carousel.js:58]: Shouldn't this be inside a `useEffect`?

**Review by ArjunGTX — APPROVED** [review]
Looks good in general. Clean and structured code. Have made some suggestions but thats not stopping me from approving!

File: frontend/src/Hooks/carousel.js (hunk 3 above)
Comment (spraveenofficial) [reply frontend/src/Hooks/carousel.js:58]: No, it's basically onClick Function.

File: frontend/src/Contexts/cart-context.js (hunk 2 above)
Comment (spraveenofficial) [reply frontend/src/Contexts/cart-context.js]: Yup doing.

**Pull request merged**